numba/cuda/cudadrv/devicearray.py

```diff
@@ -111,6 +112,24 @@ def __init__(self, shape, strides, dtype, stream=0, gpu_data=None):
         self.gpu_data = gpu_data
         self.stream = stream
 
+    @devices.require_context
+    def __getitem__(self, item):
+        return self._do_getitem(item)
+
+    def getitem(self, item, stream=0):
+        """Do `__getitem__(item)` with CUDA stream
+        """
+        return self._do_getitem(item, stream)
+
+    @devices.require_context
+    def __setitem__(self, key, value):
+        return self._do_setitem(key, value)
+
+    def setitem(self, key, value, stream=0):
```

Comment:
This should also have `@devices.require_context` for the same reason as for `getitem()`.